pkg/auth/remote_authenticator/authenticator.go

```diff
@@ -0,0 +1,164 @@
+package remote_authenticator
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/auth"
+	"github.com/treeverse/lakefs/pkg/auth/model"
+	"github.com/treeverse/lakefs/pkg/config"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+const RemoteAuthSource = "remote_authenticator"
+
+// Request object that will be sent to the remote authenticator service as JSON payload in a POST request
+type AuthenticationRequest struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// Expected response from the remote authenticator service
+type AuthenticationResponse struct {
+	// optional, if returned then the user will be used as the official username in lakeFS
+	ExternalUserIdentifier string `json:"external_user_identifier"`
+}
+
+// RemoteAuthenticator client
+type RemoteAuthenticator struct {
+	AuthService auth.Service
+	Logger      logging.Logger
+	Config      *config.RemoteAuthenticator
+	authURL     string
+	c           *http.Client
+}
+
+func NewRemoteAuthenticator(conf *config.RemoteAuthenticator, authService auth.Service, logger logging.Logger) (auth.Authenticator, error) {
+	authUrl, err := url.JoinPath(conf.BaseURL, conf.AuthEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	logger.Debugf("initializing remote authenticator with auth url %s", authUrl)
```

Comment:
Parameters should go in `WithField` rather than interpolation.

Reply:
done